src/components/AppSettingsModal/AppSettingsModal.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import produce from 'immer';
+
+import * as actions from '../../actions';
+import { getDefaultProjectPath } from '../../reducers/app-settings.reducer';
+import { COLORS } from '../../constants';
+import { setNested } from '../../utils';
+
+import Modal from '../Modal';
+import ModalHeader from '../ModalHeader';
+import Spacer from '../Spacer';
+import { FillButton } from '../Button';
+import FormField from '../FormField';
+import Toggle from '../Toggle';
+
+import DirectoryPicker from '../DirectoryPicker';
+import ProjectTypeSelection from '../ProjectTypeSelection';
+
+import type { AppSettings } from '../../types';
+
+type Props = {
+  isVisible: boolean,
+  settings: AppSettings,
+  hideModal: () => void,
+  saveAppSettings: (newSettings: AppSettings) => void,
+};
+
+type State = {
+  newSettings: AppSettings,
+};
+
+class AppSettingsModal extends PureComponent<Props, State> {
+  state = {
+    newSettings: this.props.settings,
+  };
+
+  saveSettings = (ev: SyntheticEvent<*>) => {
+    ev.preventDefault();
+
+    const { saveAppSettings } = this.props;
+    const { newSettings } = this.state;
+
+    saveAppSettings(newSettings);
+  };
+
+  // Update setting by key
+  // Notice: Key will be nested e.g. 'general.defaultProjectPath' that's why we're using utility function setNested
+  updateSetting = (key: string, value: any) => {
+    this.setState(state =>
+      // produce could be used with currying here - so no state passed to produce. But Flow couldn't detect the right return type.
+      produce(state, draftState => {
+        setNested(draftState.newSettings, key, value);
+      })
+    );
+  };
+
+  render() {
+    const { hideModal, isVisible } = this.props;
+    const newSettings = this.state.newSettings;
+    return (
+      <Modal isVisible={isVisible} onDismiss={hideModal}>
+        <ModalHeader title="Preferences" />
+
+        <MainContent>
+          <form onSubmit={this.saveSettings}>
+            {/* // todo: Refactor to map over settings so this renders dynamically -
+                         needs additional info on state e.g. component for rendering + props
+                         -> For now it's OK to hard code as we're only having three
+                            settings 
+            */}
+            <SectionTitle>General</SectionTitle>
+            <FormField label="Default Project Path" focusOnClick={false}>
+              <Spacer size={5} />
+              <DirectoryPicker

```

Comment:
I think it makes sense to leave the picker as is in the project creation flow. I'm not sure an editable input would fit there.

I like how this looks after these updates! Should we match the style with the + and the round border like this?

<img width="111" alt="screen shot 2018-11-01 at 11 05 37 am" src="https://user-images.githubusercontent.com/17421347/47870187-293e1e00-ddc6-11e8-9a4b-eaa5cb463960.png">

Also, would it be possible to have the underline stretch underneath the icon? (pull the icon to sit inside the input itself or make it seem like it's doing that). Aka mirror this:

<img width="373" alt="screen shot 2018-11-01 at 11 07 43 am" src="https://user-images.githubusercontent.com/17421347/47870263-61ddf780-ddc6-11e8-9786-976b1fe88a26.png">

Which I like better because it makes it obvious to the user that the icon and the input are both part of the same flow, and not two separate things.

Reply:
Yes, that's better. I'll add the circle with folder-plus icon and strech the line to have a similar look like in the project creation wizard.